src/index.js

```diff
@@ -486,17 +466,11 @@ class ImageMinimizerPlugin {
         compilation.hooks.assetPath.tap(
           { name: pluginName },
           (filename, data, info) => {
-            const newInfo =
-              data &&
-              // @ts-ignore
-              data.module &&
-              // @ts-ignore
-              data.module.buildMeta &&
-              // @ts-ignore
-              data.module.buildMeta.imageMinimizerPluginInfo;
+            // @ts-ignore
+            const newInfo = data?.module?.buildMeta?.imageMinimizerPluginInfo;
 
-            if (newInfo) {
-              Object.assign(info || {}, newInfo);
+            if (info && newInfo) {
```

Comment:
It changes logic

Reply:
No, if `info` does not exist (undefined) then you don't need to assign properties to new empty object  because you don't use result of this assigment.